sql/src/main/java/io/crate/operation/scalar/ArrayCatFunction.java

```diff
@@ -0,0 +1,131 @@
+        ArrayType arrayType = (ArrayType) types.get(0);
+        if(arrayType.innerType().equals(DataTypes.UNDEFINED)){
+            arrayType = (ArrayType) types.get(1);
+        }
+        return new FunctionInfo(new FunctionIdent(NAME, types), arrayType);
+    }
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(NAME, new Resolver());
+    }
+
+    protected ArrayCatFunction(FunctionInfo functionInfo) {
+        this.functionInfo = functionInfo;
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return functionInfo;
+    }
+
+    @Override
+    public Symbol normalizeSymbol(Function symbol) {
+        if (anyNonLiterals(symbol.arguments())) {
+            return symbol;
+        }
+
+
+        Input[] argList = new Input[symbol.arguments().size()];
+        int counter = 0;
+        for(Symbol argSymbol : symbol.arguments()){
+            Input input  = (Input) argSymbol;
+            argList[counter++] = input;
+        }
+
+        return Literal.newLiteral(evaluate(argList), this.info().returnType());
+    }
+
+    @Override
+    public Object[] evaluate(Input[] args) {
+        int counter = 0;
+        for(Input array : args){
+            if(array == null || array.value() == null){
+                continue;
+            }
+            Object[] arg = (Object[]) array.value();
+            counter += arg.length;
+        }
+
+        DataType innerType = ((ArrayType) this.info().returnType()).innerType();
+
+        Object[] resultArray = new Object[counter];
+        counter = 0;
+        for(Input array : args){
+            if(array == null || array.value() == null){
+                continue;
+            }
+            Object[] arg = (Object[]) array.value();
+            for(Object element: arg){
+                resultArray[counter++] = innerType.value(element);
+            }
+        }
+
+        return resultArray;
+    }
+
+
+    private static class Resolver implements DynamicFunctionResolver {
+
+        @Override
+        public FunctionImplementation<Function> getForTypes(List<DataType> dataTypes) throws IllegalArgumentException {
+            Preconditions.checkArgument(dataTypes.size() == 2, "array_cat function requires 2 arguments");
+
+            for (int i = 0; i < dataTypes.size(); i++) {
+                Preconditions.checkArgument(dataTypes.get(i) instanceof ArrayType, String.format(Locale.ENGLISH,
+                        "Argument %d of the array_cat function can't be converted to array", i+1));
+            }
+
+            DataType innerType0 = ((ArrayType) dataTypes.get(0)).innerType();
+
+            if(!innerType0.equals(DataTypes.UNDEFINED)){
+                DataType innerType1 = ((ArrayType) dataTypes.get(1)).innerType();
+                Preconditions.checkArgument(innerType1.isConvertableTo(innerType0),
+                        "Second argument's inner type of the array_cat function can't be converted to the first argument's inner type");
```

Comment:
consider adding the inner types to your log message sp the user knows what exactly is not convertable